lua/utils.lua

```diff
@@ -72,16 +72,27 @@ Module.keys = function(table_)
   return array
 end
 
-Module.split = function(string_, separator)
+Module.split_once = function(string_, separator)
   local array = {}
-
-  for s in string.gmatch(string_, "([^" .. separator .. "]+)") do
-    table.insert(array, s)
+  local start_index, end_index = string.find(string_, separator, 1, true)
+
+  if start_index then
+    -- First part before the separator
+    table.insert(array, string.sub(string_, 1, start_index - 1))
+    -- Second part after the separator
```

Comment:
plea

You did a great job with naming variables, so I think we can remove the comments:

```suggestion
    table.insert(array, string.sub(string_, 1, start_index - 1))
```